Sources/Segment/Plugins/SegmentDestination.swift

```diff
@@ -120,20 +120,25 @@ public class SegmentDestination: DestinationPlugin {
             for url in data {
                 analytics.log(message: "Processing Batch:\n\(url.lastPathComponent)")
                 
-                let uploadTask = httpClient.startBatchUpload(writeKey: analytics.configuration.values.writeKey, batch: url) { (result) in
+                let uploadTask = httpClient.startBatchUpload(writeKey: analytics.configuration.values.writeKey, batch: url) { [weak self] (result) in
                     switch result {
                         case .success(_):
                             storage.remove(file: url)
-                            self.cleanupUploads()
-                        default:
-                            analytics.logFlush()
+                            self?.cleanupUploads()
+                        case let .failure(error):
+                            // Until https://github.com/segmentio/analytics-swift/issues/152 is resolved we should remove failing files since we run the risk of building up a huge list of pending files to upload.
+                            if error.connectivityError {
+                                storage.remove(file: url)
+                            }
                     }
+
+                    analytics.logFlush()
```

Comment:
Do we wanna call this all the time now? (before it was only in the `default` case it seems?)

Reply:
Oh - I didn't get there was an upstream fix already. Thought this is gonna be it :-) 
I'm a bit confused now why we're not picking the upstream commits into this, but do our own version of it. 
Also, mind that our connectivityError's are different - `e.g.` ``timedOut` counts as no `connectivityError` upstream, where as we flag it as one. Not sure if on purpose?

The upstream PR does not fix the issue of files piling up though - does it? Reading the PR description it's just about missing the 30s flushs? 

What about pulling the upstream PR in (in case we want it now already) - but independently of it, drop any excess files lying around (with something like the approach in the other comment). And maybe also upstream that?

Reply:
Happy to pull in the upstream PR once it's landed, but as you've pointed out it's fixing something a little different than this. But we will integrate it once it's in main